opencti-platform/opencti-front/src/private/components/settings/Users.tsx

```diff
@@ -0,0 +1,249 @@
+            orderBy: $orderBy
+            orderMode: $orderMode
+            filters: $filters
+          ) @connection(key: "Pagination_users") {
+            edges {
+              node {
+                id
+                name
+                firstname
+                lastname
+                ...UsersLine_node
+              }
+            }
+            pageInfo {
+              endCursor
+              hasNextPage
+              globalCount
+            }
+          }
+        }
+      `;
+
+const usersLineFragment = graphql`
+    fragment UsersLine_node on User {
+      id
+      name
+      user_email
+      firstname
+      external
+      lastname
+      entity_type
+      effective_confidence_level {
+        max_confidence
+      }
+      otp_activated
+      created_at
+    }
+  `;
+
+const LOCAL_STORAGE_KEY = 'users';
+
+const Users = () => {
+  const isEnterpriseEdition = useEnterpriseEdition();
+  const { t_i18n } = useFormatter();
+  const { setTitle } = useConnectedDocumentModifier();
+  setTitle(t_i18n('Users | Security | Settings'));
+  const isSetAccess = useGranted([SETTINGS_SETACCESSES]);
+  const isAdminOrganization = useGranted([VIRTUAL_ORGANIZATION_ADMIN]);
+  const { me } = useAuth();
+  const organization = me.administrated_organizations?.[0] ?? null;
+
+  const initialValues = {
+    searchTerm: '',
+    sortBy: 'name',
+    orderAsc: true,
+    openExports: false,
+    filters: emptyFilterGroup,
+  };
+
+  const {
+    viewStorage,
+    paginationOptions,
+    helpers,
+  } = usePaginationLocalStorage<UsersLinesPaginationQuery$variables>(
+    LOCAL_STORAGE_KEY,
+    initialValues,
+  );
+
+  const { filters } = viewStorage;
+  const contextFilters = useBuildEntityTypeBasedFilterContext('User', filters);
+
+  const queryPaginationOptions = {
+    ...paginationOptions,
+    filters: contextFilters,
+  } as unknown as UsersLinesPaginationQuery$variables;
+
+  const queryRef = useQueryLoading<UsersLinesPaginationQuery>(
+    usersQuery,
+    queryPaginationOptions,
+  );
+
+  const defaultAssignationFilter = {
+    mode: 'and',
+    filters: [{ key: 'default_assignation', values: [true] }],
+    filterGroups: [],
+  };
+  const defaultGroupsQueryRef = useQueryLoading(
+    groupsQuery,
+    {
+      orderBy: 'name',
+      orderMode: 'asc',
+      filters: defaultAssignationFilter,
+    },
+  );
+
+  let userCreateButton;
+  if (isSetAccess && defaultGroupsQueryRef) {
+    userCreateButton = (
+      <React.Suspense fallback={<Loader variant={LoaderVariant.inElement} />}>
+        <UserCreation paginationOptions={queryPaginationOptions} defaultGroupsQueryRef={defaultGroupsQueryRef} />
+      </React.Suspense>
+    );
+  } else if (!isSetAccess && isAdminOrganization && isEnterpriseEdition) {
+    userCreateButton = (
+      <SettingsOrganizationUserCreation
+        paginationOptions={queryPaginationOptions}
+        organization={organization}
+        variant="controlledDial"
+      />
+    );
+  }
```

Comment:
The **Suspense fallback** cause some flash/height issue during loading, we don't really the fallback. This is enough:
```
      <React.Suspense>
```

It could also be good to wrap this in a `useMemo` since isSetAccess, isAdminOrganization, etc. change rarely and it might be lighter for the query loading (and also using immutable :smile:)
```suggestion
const userCreateButton = useMemo(() => {
  if (isSetAccess && defaultGroupsQueryRef) {
    return (
      <React.Suspense fallback={<Loader variant={LoaderVariant.inElement} />}>
        <UserCreation 
          paginationOptions={queryPaginationOptions} 
          defaultGroupsQueryRef={defaultGroupsQueryRef} 
        />
      </React.Suspense>
    );
  }
  if (!isSetAccess && isAdminOrganization && isEnterpriseEdition) {
    return (
      <SettingsOrganizationUserCreation
        paginationOptions={queryPaginationOptions}
        organization={organization}
        variant="controlledDial"
      />
    );
  }
  return null;
}, [
  isSetAccess, 
  defaultGroupsQueryRef, 
  isAdminOrganization, 
  isEnterpriseEdition,
  queryPaginationOptions,
  organization
]);```